src/bp/nlu-server/validation/extractVariables.ts

```diff
@@ -0,0 +1,17 @@
+export default function extractVariables(text: string) {
+  // check out unit test to see the regex power
+  const variableRegex = /\$[A-Za-z][\w-]*/gm
```

Comment:
needs top start with $ so just add ^ and you'll be even more powerful

`/^\$[a-zA-Z]+/`

(you'll might have to split by space though...) is that what you're trying to avoid

Reply:
Well I don't want to force the user to add a space before a variable... Is there even spaces in non-latin languages? Even in english, I'm just trying to avoid this:

"I want to buy $fruit, $fruit,$fruit and milk."
<--------------------------------^^^^^^----------->

OR

"Please buy the following item:$fruit"

Also, the augmenter doesnt require that variables are separated with spaces, so I don't see why this extractor should. 

To me, the best solution to all problems regarding variables is to allow the user to escape a dollar sign symbol. IMO, a dollar sign, no matter where it is, should indicate the start of a variable even if the variable name is invalid. We should then force the user to escape his dollar sign if he's not trying to reference a variable. This logic is dumb and easely maintainable in all languages.

What do you think?

Reply:
let's adress this problem another time!